tests/spring-cloud-sleuth-instrumentation-reactor-tests/src/test/java/org/springframework/cloud/sleuth/instrument/web/client/WebClientBraveTests.java

```diff
@@ -107,14 +99,12 @@ public void onSubscribe(Subscription s) {
 			}
 
 			@Override
-			public void onNext(ClientResponse t) {
-				Subscription s = ref.getAndSet(null);
-				if (s != null) {
+			public void onNext(ClientResponse clientResponse) {
+				if (ref.getAndSet(null) != null) {
 					callback.onSuccess(null);
-					s.cancel();
 				}
 				else {
-					Operators.onNextDropped(t, currentContext());
+					Operators.onNextDropped(clientResponse, currentContext());
```

Comment:
again, with a `Mono` source this should never happen... maybe make it fail the test rather than `onNextDropped` (which by default only logs, IIRC).

Reply:
sounds good